File: benchmarks/cpu/divsum/divsum.go
package main

import "fmt"

func divsum(n int) int64 {
	var total int64
	for i := 1; i <= n; i++ {
		var s int64
		for j := 1; j <= i; j++ {
			if i%j == 0 {
				s += int64(j)
			}
		}
		total += s
	}
	return total
}

func main() {
	fmt.Println(divsum(10000))
}

File: benchmarks/cpu/divsum/divsum.ml
let divsum n =
  let total = ref 0L in
  for i = 1 to n do
    let s = ref 0L in
    for j = 1 to i do
      if i mod j = 0 then s := Int64.add !s (Int64.of_int j)
    done;
    total := Int64.add !total !s
  done;
  !total

let () = Printf.printf "%Ld\n" (divsum 10000)
